addons/lunch/models/lunch.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+
+import datetime
+from openerp import models, fields, api, _
+
+class lunch_order(models.Model):
+    """
+    lunch order (contains one or more lunch order line(s))
+    """
+    _name = 'lunch.order'
+    _description = 'Lunch Order'
+    _order = 'date desc'
+
+    @api.multi
+    def name_get(self):
+        return [(order.id, _("Lunch Order #%d") % order.id) for order in self]
+
+    def _default_alerts_get(self):
+        """
+        get the alerts to display on the order form
+        """
+        alert_msg = [alert.get_alert_message()
+                     for alert in self.env['lunch.alert'].search([])
+                     if alert.get_alert_message()]
+        return '\n'.join(alert_msg)
+
+    def _compute_alerts_get(self):
+        """
+        get the alerts to display on the order form
+        """
+        if self.state == 'new':
+            self.alerts = self._default_alerts_get()
+
+    def _default_get_previous_order_ids(self):
```

Comment:
If I get rid of the default, previous_order_ids are not set, so no previous order is displayed. When I save, however, the previous orders are displayed since the field is calculated. Again, seems I need 2 methods :/